### ISSUE TITLE
Files with trailing dots break on Windows

### ISSUE THREAD

**Issue by spf13** [nushell/nushell#7869]
### Describe the bug

While the Win32 API doesn't allow files or folders with trailing dots, the filesystems do allow it and programs do and can create files with trailing dots or spaces. 

If this happens nushell reports the following

Error: nu::shell::error_reading_file (link)

```
  × Error trying to read file
   ╭─[entry #21:1:1]
 1 │ ls
   · ─┬
   ·  ╰── Could not read file metadata
   ╰────
```

### How to reproduce

In Bash on Windows 
touch "foo."

In Nushell on Windows 
ls

### Expected behavior

I expected nu to list the directory regardless of files that end in spaces or dots.... just like bash does. 

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.71.0                                          |
| branch             |                                                 |
| commit_hash        | 6cc4ef6c706ebbdc5b995013e0b00f77f7a4dba7        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.65.0 (897e37553 2022-11-02)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.65.0 (4bc8f24d3 2022-10-20)             |
| pkg_version        | 0.71.0                                          |
| build_time         | 2022-11-08 18:03:10 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by ChrisDenton** [comment]
This is fundamentally a consequence of how Win32 APIs work. They strip trailing dots from paths when called.

MINGW bash uses lower level (NT) APIs that don't do this. It's also possible to use verbatim paths to workaround this issue.

**Comment by ChrisDenton** [comment]
I guess the `ls` command specifically could be improved to not error in this case? E.g. if the file's metadata can't be retrieved for whatever reason it could have empty cells in the table or it could make use of whatever metadata is returned from `fs::read_dir` (I believe this is currently discarded by nu-glob).

**Comment by rgwood** [comment]
The error message is better in recent versions of Nu: https://github.com/nushell/nushell/pull/7348

It would be nice to handle this as well as MINGW bash but tbh it's not a high priority; IIRC Windows Explorer/cmd/PowerShell don't even show these files.

**Comment by spf13** [comment]
cmd, Windows Explorer and PowerShell all show these files. They struggle
to open, move or rename the file/directory, but Windows explorer even lets
you show the properties of it.

It's odd that Nu errors instead of listing.

**Comment by rgwood** [comment]
Yeah, I corrected my comment shortly after posting (currently travelling and on very little sleep)

**Comment by rgwood** [comment]
If anyone wants to tackle this, this is the relevant place in `ls`: https://github.com/nushell/nushell/blob/76292ef10c6eb4c3eaaba7cf1ee39348580bc55f/crates/nu-command/src/filesystem/ls.rs#L688-L691

It's pretty easy to give up and return nothing for metadata columns when `FindFirstFileW` fails, here's something I hacked together: ![image](https://user-images.githubusercontent.com/26268125/215119636-374dabf5-0942-4830-ab3d-5e2e1ef8a8f6.png)

That might be good enough as a quick-and-dirty improvement until someone has time to figure out a second fallback mechanism using Chris's suggestion:

> MINGW bash uses lower level (NT) APIs that don't do this. It's also possible to use verbatim paths to workaround this issue.

I'm on vacation with a slow laptop but I will aim to put up a PR in a few weeks if nobody gets around to it earlier.

**Comment by rgwood** [comment]
I've just merged a fix that handles this a little better: https://github.com/nushell/nushell/pull/7999

![image](https://user-images.githubusercontent.com/26268125/217358478-bdc60e65-e8ff-48fd-80f4-935d12c72a67.png)

Eventually we'd like to get to a place where Nu can read the metadata of files with 'illegal' names too. Chris's suggestion of verbatim paths seems like the most promising approach, IMO. Might as well leave this issue open until we get around to the full solution.
